src/components/Modals/NewMessageModal.jsx

```diff
@@ -0,0 +1,220 @@
+// React Imports
+import React, { useState, useContext } from 'react';
+// Inrupt Library Imports
+import { useSession } from '@hooks';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { sendMessageTTL, getMessageTTL } from '../../utils';
+// Context Imports
+import { MessageContext, SignedInUserContext } from '../../contexts';
+
+/**
```

Comment:
There seems to be a mismatch between JSDoc here and JSDoc in Development.

Assuming this was supposed to be a copy of `NewMessage`, there might have been a problem during conflict resolution or git rebase. The typedef for the component prop for this component has been moved to typedefs.js. On top of that, the props have changed for this component, so we should update that.

Reply:
How should it be updated? I haven't messed with the `typedefs.js` file yet.

Reply:
The JSDoc over in Development for `NewMessage.jsx` looks like this:

```
/**
 * @typedef {import("../../typedefs.js").newMessageProps} newMessageProps
 */

/**
 * New Message Component - Component that allows user to write
 *  a message to another user from their inbox
 *
 * @memberof Inbox
 * @name NewMessage
 * @param {newMessageProps} Props - Props used for NewMessage
 * @returns {React.JSX.Element} React component for NewMessage
 */
```

I think you could simply copy and paste it in there.

Reply:
Alright I've copied it over. Is this alright? Or should it just be these two and not include this:
```
/**
 * messageFolderProps is an object that stores the props for the MessageFolder
 * component
 *
 * @typedef newMessageModalProps
 * @type {object}
 * @property {() => void} closeForm - The function used to trigger NewMessageModal to
 * close
 * @memberof typedefs
 */
```

Reply:
You would have to update the JSDoc typedefs for the props since you're passing in different one from what the component uses originally

Reply:
Do we still need the `closeForm` function though? Isn't using MUI Modals making that unnecessary?

Reply:
You would need to include `showModal`, `setShowModal`, and `oldMessage` as `@property` inside `newMessageProps` in `typedefs.js` in place of `closeForm`. You would also have to rename the typedef to `newMessageModalProps` to match with the type being used to define the props in the modal.

From there you would simply need to import `newMessageModalProps` from that file as shown from my earlier comment. You don't need to define the typedef again in the modal.

On top of that, it would seem there's a typo in the description for `newMessageProps`, which should be `newMessageModalProps` and not `messageFoldProps` and it should be props for the `NewMessageModal` component, not the `MessageFolder` component.

Reply:
Alright, this should be done now. Fingers crossed.